internal/provider/resource_gitlab_project_environment.go

```diff
@@ -0,0 +1,224 @@
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"stop_before_destroy": {
+				Description: "Determines whether the environment is attempted to be stopped before the environment is deleted.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+			},
+			"created_at": {
+				Description: "The date/time that this environment was created at in Zulu time (UTC)",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"updated_at": {
+				Description: "The date/time that this environment was last updated in Zulu time (UTC)",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"state": {
+				Description: fmt.Sprintf("State the environment is in. Valid values are %s.", renderValueListForDocs(validProjectEnvironmentStates)),
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+})
+
+func resourceGitlabProjectEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	options := gitlab.CreateEnvironmentOptions{
+		Name: &name,
+	}
+	if externalURL, ok := d.GetOk("external_url"); ok {
+		options.ExternalURL = gitlab.String(externalURL.(string))
+	}
+
+	project := d.Get("project").(string)
+
+	log.Printf("[DEBUG] Project %s create gitlab environment %q", project, *options.Name)
+
+	client := meta.(*gitlab.Client)
+
+	environment, _, err := client.Environments.CreateEnvironment(project, &options, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			return diag.Errorf("feature Environments is not available")
+		}
+		return diag.FromErr(err)
+	}
+
+	environmentID := fmt.Sprintf("%d", environment.ID)
+	d.SetId(buildTwoPartID(&project, &environmentID))
+
+	return resourceGitlabProjectEnvironmentRead(ctx, d, meta)
+}
+
+func resourceGitlabProjectEnvironmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] read gitlab environment %s", d.Id())
+
+	project, environmentIDInt, err := environmentAndProjectFromResource(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] Project %s read gitlab environment %d", project, environmentIDInt)
+
+	client := meta.(*gitlab.Client)
+
+	environment, _, err := client.Environments.GetEnvironment(project, environmentIDInt, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			log.Printf("[DEBUG] Project %s gitlab environment %q not found, removing from state", project, environmentIDInt)
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error getting gitlab project %q environment %q: %v", project, environmentIDInt, err)
+	}
+
+	d.Set("project", project)
+	d.Set("name", environment.Name)
+	d.Set("state", environment.State)
+	d.Set("external_url", environment.ExternalURL)
+
```

Comment:
Missing `created_at` and `updated_at` in state. Also format them as `time.RFC3339` ;)

Reply:
```suggestion
	d.Set("created_at", environment.CreatedAt.Format(time.RFC3339))
	if environment.UpdatedAt != nil {
		d.Set("updated_at", environment.UpdatedAt.Format(time.RFC3339))
	}
```

Reply:
For some reason I thought that computed fields were automatically set in state thank you for putting down the format to import them :)

Reply:
Implemented